constraint/pkg/client/drivers/local/driver.go

```diff
@@ -287,25 +296,45 @@ func (d *Driver) Query(ctx context.Context, target string, constraints []*unstru
 
 		kindResults, err := drivers.ToResults(constraintsMap, resultSet)
 		if err != nil {
-			return nil, nil, err
-		}
-
-		for _, result := range kindResults {
-			result.EvaluationMeta = RegoEvaluationMeta{
-				TemplateRunTime: float64(evalEndTime.Nanoseconds()) / 1000000,
-				ConstraintCount: uint(len(kindResults)),
-			}
+			return &drivers.QueryResponse{Err: err}
 		}
 
 		results = append(results, kindResults...)
+
+		if d.gatherMetrics {
+			// for the batch of evaluated constraints, report their evaluation metadata
+			evalRunTime := float64(evalEndTime.Nanoseconds()) / 1000000
```

Comment:
What is the impact on serialization/deserialization of float vs. int?

Reply:
are you asking in terms of performance?

Reply:
In terms of accuracy and consistency. In general floats are harder to round-trip accurately:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

Reply:
Interesting, good points.

One problem with using `ints` that I ran into before when gauging duration of a function is when integer division kicks in for values < 1. So evaluations that take less than a millisecond, in our case, can come out as `0`. In practice, I understand this may not be an issue but, theoretically, reporting `[0-1)` values as `0` could happen. 

Wdyt? If we can live with this edge case, I'm happy to use an int here

Reply:
Per our offline chat, I've started to use nanoseconds to get around the issue above!